src/cdomains/baseDomain.ml

```diff
@@ -97,14 +97,6 @@ struct
     let tr = QCheck.quad (CPA.arbitrary ()) (PartDeps.arbitrary ()) (WeakUpdates.arbitrary ()) (PrivD.arbitrary ()) in
     QCheck.map ~rev:to_tuple of_tuple tr
 
-  let bot () = { cpa = CPA.bot (); deps = PartDeps.bot (); weak = WeakUpdates.bot (); priv = PrivD.bot ()}
-  let is_bot {cpa; deps; weak; priv} = CPA.is_bot cpa && PartDeps.is_bot deps && WeakUpdates.is_bot weak && PrivD.is_bot priv
-  let top () = {cpa = CPA.top (); deps = PartDeps.top ();  weak = WeakUpdates.top () ; priv = PrivD.bot ()}
```

Comment:
I think the generated version will be different w.r.t. to `priv` here, so we might have to leave this one!

Reply:
I'm pretty sure that's a copy-paste typo because on the following line, it's all tops:
```ocaml
let is_top {cpa; deps; weak; priv} = CPA.is_top cpa && PartDeps.is_top deps && WeakUpdates.is_top weak && PrivD.is_top priv
```
Or do you know of a reason why it should be that way? It doesn't make sense to me that it's `PrivD.bot` in both `bot` and `top` because that could only be if it's a unit lattice.

So I think this is the precise reason to have a PPX deriver to avoid such typos.

Reply:
I think the idea is that even in top state, there's nothing privatized in the hopes that some additional precision can be regained later. Do the regression tests still work? And if yes, how about the tests we ran for traces and more-traces? Does this change the results there?

Reply:
The hand-written version was fixed in #1463.